### PR TITLE
feat: flattening getReceiver call

### PR DESCRIPTION
**Pull request by manishdangi98**

### CODE REVIEW

File: core/receiver/receiver.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 type Repository interface {
 	List(context.Context, Filter) ([]Receiver, error)
 	Create(context.Context, *Receiver) error
-	Get(context.Context, uint64) (*Receiver, error)
+	Get(context.Context, uint64, bool) (*Receiver, error)
```
Comment (mabdh) [inline core/receiver/receiver.go:12]: we don't need to keep the old behaviour

File: internal/store/model/receiver.go [hunk 2]
```diff
@@ -20,6 +20,19 @@ type Receiver struct {
...
+	Name                 string              `db:"name"`
+	Type                 string              `db:"type"`
+	Labels               pgc.StringStringMap `db:"labels"`
+	Configurations       pgc.StringAnyMap    `db:"configurations"`
+	Data                 pgc.StringAnyMap    `db:"-"` //TODO do we need this?
+	ParentID             sql.NullInt64       `db:"parent_id"`
+	CreatedAt            time.Time           `db:"created_at"`
+	UpdatedAt            time.Time           `db:"updated_at"`
+	ParentConfigurations pgc.StringAnyMap    `db:"parent_configurations"`
+}
```
Comment (mabdh) [inline internal/store/model/receiver.go:34]: Why do we need to create another model here?
We could just put the merged configurations in `configurations` right?

File: internal/store/postgres/receiver.go [hunk 3]
```diff
@@ -41,6 +41,18 @@ var receiverListQueryBuilder = sq.Select(
...
+var receiverListLeftJoinSelfQueryBuilder = sq.Select(
+	"receiver.id",
+	"receiver.name",
+	"receiver.type",
+	"receiver.labels",
+	"receiver.configurations",
+	"receiver.parent_id",
+	"receiver.created_at",
+	"receiver.updated_at",
+	"parent_receiver.configurations AS parent_configurations",
```
Comment (mabdh) [inline internal/store/postgres/receiver.go:53]: the configurations could be merged and returned in a single column.

please check how to merge two json in psql

File: internal/store/postgres/receiver.go [hunk 4]
```diff
@@ -41,6 +41,18 @@ var receiverListQueryBuilder = sq.Select(
...
+).From("receivers receiver")
```
Comment (mabdh) [inline internal/store/postgres/receiver.go:54]: This is confusing, just keep it simple

```suggestion
).From("receivers r")
```

File: internal/store/postgres/receiver_test.go [hunk 5]
```diff
@@ -229,17 +244,40 @@ func (s *ReceiverRepositoryTestSuite) TestGet() {
...
 			PassedID:    1000,
 			ErrString:   "receiver with id 1000 not found",
+			WithExpand:  false,
+		},
+		{
+			Description: "should get a receiver with parent configurations when with expand set to true",
+			PassedID:   4,
+			ExpectedReceiver: &receiver.Receiver{
+				ID:   4,
+				Name: "de-fabric-slack",
```
Comment (mabdh) [inline internal/store/postgres/receiver_test.go:260]: please use another name that is more generic and not internal name

File: core/receiver/receiver.go (hunk 1 above)
Comment (manishdangi98) [reply core/receiver/receiver.go:12]: Didn't got this can you explain this further

Comment (mabdh) [reply core/receiver/receiver.go:12]: Let me start with question. Probably I understand it wrong.

Why we need to add bool here?

Comment (manishdangi98) [reply core/receiver/receiver.go:12]: With this bool here we are defining weather we want to expand the get call or not.

Comment (mabdh) [reply core/receiver/receiver.go:12]: In that case, lets use filter like the one being used by List?

File: internal/store/model/receiver.go [hunk 6]
```diff
@@ -20,6 +20,19 @@ type Receiver struct {
 	UpdatedAt      time.Time           `db:"updated_at"`
 }
 
+type ExpandedReceiver struct {
```
Comment (ishanarya0) [inline internal/store/model/receiver.go:23]: lets remove this?

**Review by ishanarya0 — CHANGES_REQUESTED** [review]

File: internal/store/model/receiver.go (hunk 6 above)
Comment (mabdh) [inline internal/store/model/receiver.go:23]: This could be removed I guess?

**Review by ishanarya0 — APPROVED** [review]

**Pull request merged**